### PR TITLE
Fix Statcache issue

### PR DESCRIPTION
**Pull request by kobergj**
Statcache is not working correctly. Some tests fail in local but are green on ci.

Goals:

1) Make tests red on ci
2) Make tests green on ci

Note: The order is important

### CODE REVIEW

**Comment by kobergj** [comment]
@rhafer this should fix the (locally) failing test in ci and on local. Please double check!

File: internal/grpc/services/gateway/storageprovidercache.go [hunk 1]
```diff
@@ -86,6 +101,11 @@ func (c *cachedAPIClient) Stat(ctx context.Context, in *provider.StatRequest, op
 		return resp, nil
 	case key == "":
 		return resp, nil
+	case strings.Contains(key, "sid:a0ca6a90-a365-4782-871e-d44447bbc668"):
```
Comment (rhafer) [inline internal/grpc/services/gateway/storageprovidercache.go:104]: Shouldn't this ID be defined as a constant in some central place? Having the hardcoded ID here (and elsewhere) makes the code a) harder to understand and b) makes it hard to keep stuff in sync in the code.

**Review by rhafer — CHANGES_REQUESTED** [review]
> this should fix the (locally) failing test in ci and on local. Please double check!

Yeah, this works locally here now was well. Thanks. 

This looks good to me overall. Though I would really like to avoid adding the hardcoded spaceid here. Is there a way we can move that into a constant?

File: internal/grpc/services/gateway/storageprovidercache.go (hunk 1 above)
Comment (kobergj) [reply internal/grpc/services/gateway/storageprovidercache.go:104]: Good point. The question is: Where would that constant live? For now it is created within the sharestorageprovider, see here: https://github.com/cs3org/reva/blob/edge/internal/grpc/services/sharesstorageprovider/sharesstorageprovider.go#L352

Ofc it is no problem to move it to a constant in the gateway package, but if we want to use the same constant in sharesstorageprovider and gateway we need a common package that both can import...

Comment (rhafer) [reply internal/grpc/services/gateway/storageprovidercache.go:104]: Good question ;-). How about pkg/utils? It's not really nice, but it  should work...

Comment (refs) [reply internal/grpc/services/gateway/storageprovidercache.go:104]: There was another magic string like this I recently found. If we start setting default id's without a formal configuration then things will get confusing real fast. This is the one I refer to:

https://github.com/cs3org/reva/blob/edge/internal/grpc/services/publicstorageprovider/publicstorageprovider.go#L53

Comment (kobergj) [reply internal/grpc/services/gateway/storageprovidercache.go]: @refs @rhafer I moved both to utils package and import them from there. Should be easy to add configuration if we see fit. For now if you really need to change the IDs you can overwrite them via build flags

Comment (refs) [reply internal/grpc/services/gateway/storageprovidercache.go]: 🙆‍♂️ Thanks! :D

**Review by rhafer — APPROVED** [review]

**Pull request merged**